File: typescript-client-sdk/src/sdk/models/shared/integration.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";


export enum IntegrationIntegrationIdEnum {
    Salesforce = "salesforce",
    Hubspot = "hubspot",
    Pipedrive = "pipedrive"
}


export class Integration extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=iconURL" })
  iconURL?: string;

  @SpeakeasyMetadata({ data: "json, name=integrationId" })
  integrationId?: IntegrationIntegrationIdEnum;

  @SpeakeasyMetadata({ data: "json, name=name" })
  name?: string;
}

File: python-client-sdk/src/orbapi/models/shared/integration.py
import dataclasses
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from orbapi import utils

class IntegrationIntegrationIDEnum(str, Enum):
    SALESFORCE = "salesforce"
    HUBSPOT = "hubspot"
    PIPEDRIVE = "pipedrive"


@dataclass_json
@dataclasses.dataclass
class Integration:
    icon_url: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('iconURL') }})
    integration_id: Optional[IntegrationIntegrationIDEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('integrationId') }})
    name: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('name') }})
    
